mono/mini/interp/transform.c

```diff
@@ -1046,9 +1046,8 @@ static void
 interp_generate_ipe_throw_with_msg (TransformData *td, MonoError *error_msg)
 {
 	MonoJitICallInfo *info = &mono_get_jit_icall_info ()->mono_throw_invalid_program;
-	MonoMemoryManager *memory_manager = mono_domain_ambient_memory_manager (td->rtm->domain);
 
-	char *msg = mono_mempool_strdup (memory_manager->mp, mono_error_get_message (error_msg));
+	char *msg = mono_mem_manager_strdup (td->mem_manager, mono_error_get_message (error_msg));
```

Comment:
This adds locking where it wasn't present previously, is that intended?

Reply:
Having extra locking is only a problem during steady state execution, the transform code is part of startup.